client/parties/party-view.tsx

```diff
@@ -130,25 +129,29 @@ function renderPartyMessage(msg: Message) {
   }
 }
 
-export function PartyView() {
+interface PartyViewProps {
+  params: { partyId: string }
+}
+
+export function PartyView(props: PartyViewProps) {
   const dispatch = useAppDispatch()
   const party = useAppSelector(s => s.party)
   const partyId = party.id
-  const prevPartyId = usePrevious(partyId)
+  const routePartyId = decodeURIComponent(props.params.partyId)
   // Party can change when a user accepts an invite to a new party while already being a member of a
   // different party that is currently being rendered.
-  const partyChanged = !!prevPartyId && prevPartyId !== partyId
+  const partyChanged = routePartyId !== partyId
```

Comment:
This isn't a good naming or a description of what is occurring here, imo

Yes, changing parties is *one way* that a mismatch between the current URL and the party state can occur, but it is not the *only* way. Another way is that someone links to their own party view while you're in a different party.

So I would either A) not name this, just inline this condition in the useEffect below, since the condition is self-explanatory or B) name it something better, like `urlMatchesState`. I probably prefer A